### PR TITLE
Bug Report: Model Observer Testing

### PR DESCRIPTION
**Pull request by Mulkave**
When testing models and their observers, in the first test run in a testing class the observer methods are called as usual, but are never called for the rest of the testing methods.

### Run The Tests
This pull request contains reproduces the issue, all you have to do is:
- clone this PR
- install dependencies with `composer install`
- run the tests with `./vendor/bin/phpunit`

The result will look as follows:

screen shot 2015-07-31 at 6 04 03 pm

The constraint error occurs because the `observed` key was not set by the observer, though worked correctly in the first test.

> This is not the first time I encounter this issue and have worked around it in the past by stuffing all the test methods in the first test method and calling them manually with something like:

```php
public function test_all_tests()
{
        $this->caching_articles();
        $this->caching_article_without_author();
        $this->caching_article_without_cover();
        $this->caching_article_without_photos();
}
```

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
This was fixed in 5.1.8.

File: app/User.php [hunk 1]
```diff
@@ -33,3 +33,5 @@ class User extends Model implements AuthenticatableContract, CanResetPasswordCon
      */
     protected $hidden = ['password', 'remember_token'];
 }
+
+User::observe(new UserObserver());
```
Comment (GrahamCampbell) [inline app/User.php]: you can't put that there

Comment (GrahamCampbell) [inline app/User.php]: You MUST put that in a service provider.

Comment (GrahamCampbell) [inline app/User.php]: It's VERY bad to have this registered on class load, which is what you're doing,

Comment (GrahamCampbell) [inline app/User.php]: It also violates PSR0/PSR4, which is what you are using for autoloading.

**Comment by Mulkave** [comment]
@GrahamCampbell thanks for the indication.

If I'm understanding you correctly, you're saying that this observer setup should live in a service provider as such:

```php
public function register()
{
    App\ User::observe(new UserObserver());
}
```

**Comment by GrahamCampbell** [comment]
Boot method would be better, yes.

**Comment by GrahamCampbell** [comment]
If you put tha tin the boot method of a service provider, you should be good.

**Pull request closed without merging**